client/my-sites/plans/current-plan/index.jsx

```diff
@@ -146,19 +143,22 @@ class CurrentPlan extends Component {
 
 export default connect(
 	( state, ownProps ) => {
-		const selectedSite = getSelectedSite( state ),
-			selectedSiteId = getSelectedSiteId( state );
+		const selectedSite = getSelectedSite( state );
+		const selectedSiteId = getSelectedSiteId( state );
+
+		const isJetpack = isJetpackSite( state, selectedSiteId );
+		const isAutomatedTransfer = isSiteAutomatedTransfer( state, selectedSiteId );
 
 		return {
 			selectedSite,
 			selectedSiteId,
 			context: ownProps.context,
 			currentPlan: getCurrentPlan( state, selectedSiteId ),
 			isExpiring: isCurrentPlanExpiring( state, selectedSiteId ),
-			isJetpack: isJetpackSite( state, selectedSiteId ),
+			shouldShowDomainWarnings: ! isJetpack || isAutomatedTransfer,
+			hasDomainsLoaded: ! isRequestingSiteDomains( state, selectedSiteId ),
```

Comment:
Let's check for the existence of domains object instead. If we have persisted data we won't see a warning until we get the first response back. This will also flicker on any additional requests.